mitodl/micromasters
# Use temporary index during recreate_index

#### What are the relevant tickets?
Fixes #2550

#### What's this PR do?
Does the reindex in a separate Elasticsearch index, then deletes the old one, recreates it with a mapping and replaces it with the temp index's documents. There is still a tiny amount of time where the old index will go away and come back (less than a second for 1400 documents on my computer), and there is still a risk of stale data if documents are modified during the reindex.

This PR also updates the indexing to check if the temp index exists, then update it if it does. This is to prevent stale data in the temp index from overwriting the current one.

#### How should this be manually tested?
Run a reindex and go to `/learners`. The list of learners should not change during the reindex.
